### PR TITLE
Add bulk addition to string list component

### LINKED ISSUE #9707: pkg/rancher-components/StringList: add bulk addition
https://github.com/rancher-sandbox/rancher-desktop/issues/5344 requires a bulk addition option to be added to the string list component. The bulk addition makes it easier to pass a bunch of data to the component at once. 

Features:

* Bulk addition can be enabled and disabled
* The delimiter can be defined for each component individually
* By passing a string to the input field, the string will be split by the delimiter
* It won't accept the string (and show an error), if there is a duplicate within the string itself and also in comparison to other entries that are already on the list
* Once enabled, the bulk addition applies to the add mode as well as the edit mode

Required property:

`bulkAdditionDelimiter`: Enables bulk addition and defines the delimiter to split the input string. `null` by default.

### PR DESCRIPTION
**Pull request by s0nea**
### Summary
Fixes #9707

Add bulk addition option to string list component.

### Screenshot/Video

[rd.webm](https://github.com/rancher/dashboard/assets/8761082/5e341626-c84f-4b1e-907e-c6f3f779826e)

### CODE REVIEW

File: pkg/rancher-components/src/components/StringList/StringList.vue [hunk 1]
```diff
@@ -360,6 +356,47 @@ export default Vue.extend({
...
 
+    /**
+     * Add a new or update an exiting item in the items list.
+     *
+     * @param items The current items list.
+     * @param value The new value to be added.
+     * @param index The list index of the item to be updated (optional).
+     * @returns Updated items list.
+     */
+    addValueToItems(items: string[], value: string, index?: number): string[] {
```
Comment (torchiaf) [inline pkg/rancher-components/src/components/StringList/StringList.vue:367]: I think the `items` parameter may be decoupled from the return value here:
```
const newItems = [...items];
...
return newItems;
```
and change the input wherever `addValueToItems` is called: `addValueToItems([...this.items]) -> addValueToItems(this.items)`

File: pkg/rancher-components/src/components/StringList/StringList.vue [hunk 2]
```diff
@@ -343,12 +343,8 @@ export default Vue.extend({
...
 
       if (value) {
-        const items = [...this.items];
-        const index = findStringIndex(items, item, false);
-
-        if (index !== -1) {
-          items[index] = value;
-        }
+        const index = findStringIndex([...this.items], item, false);
+        const items = index !== -1 ? this.addValueToItems([...this.items], value, index) : [...this.items];
```
Comment (torchiaf) [inline pkg/rancher-components/src/components/StringList/StringList.vue:347]: see comment on `addValueToItems` below

File: pkg/rancher-components/src/components/StringList/StringList.test.ts [hunk 3]
```diff
@@ -398,6 +398,241 @@ describe('stringList.vue', () => {
...
+        // press enter
+        await inputField.trigger('keydown.enter');
+        await wrapper.vm.$nextTick();
+
+        const itemsResult = (wrapper.emitted('change') || [])[0][0];
+
+        expect(JSON.stringify(itemsResult)).toBe(JSON.stringify(result));
+      });
+
+      it('should show warning if one of the values is a duplicate', async() => {
```
Comment (torchiaf) [inline pkg/rancher-components/src/components/StringList/StringList.test.ts:545]: Could you please add a new `it()` block to test the case of duplicate items inside bulk items?
example:
```
const newValue = 'test;test';
```

**Review by torchiaf — CHANGES_REQUESTED** [review]
The bulkable functionality works great, just a couple of comment about possible test/code imporvements

**Comment by s0nea** [comment]
> The bulkable functionality works great, just a couple of comments about possible test/code improvements.

@torchiaf thank you for your review! I think I addressed all of your comments. Would you mind taking another look at it?

**Review by torchiaf — APPROVED** [review]
LGTM

**Review by rak-phillip — COMMENTED** [review]
@s0nea can we bump the rancher components version so that we will be ready to publish a release once this merges?

**Comment by s0nea** [comment]
> @s0nea can we bump the rancher components version so that we will be ready to publish a release once this merges?

@rak-phillip yes, thank you for mentioning it! I pushed an update.

**Review by rak-phillip — APPROVED** [review]
